### DISCUSSION TITLE
Subsurface: Let's connect the Geo-Python Stack! `t21-hack-subsurface`

### DISCUSSION THREAD

**Discussion by prisae** [softwareunderground/transform-2021-hackathon#5] [Projects]
## Slack channel
[t21-hack-subsurface](https://swung.slack.com/archives/C01S41W1LN8)

## People
* Dieter Werthmüller [TU Delft]
* Miguel de la Varga [Terranigma, **GemPy**]
* Bane Sullivan [KitWare, **PyVista**]
* Dom Fournier [Mira Geoscience, **SimPEG**]
* Santiago Soler [UN San Juan, **Fatiando**]
* Florian Wagner [RWTH, **pyGIMLi**]
* and many many more (have a look at the [subsurface](https://swung.slack.com/archives/CJLSWG9HD) channel on SWUNG, currently at 130 members)

## Description
Thanks to the awesome work of Miguel and friends the [subsurface](https://github.com/softwareunderground/subsurface) package is at a stage where it can be implemented into relevant packages (GemPy, SimPEG, PyGIMLi, Harmonica, Welly, Segysak, you-name-it). What started at **t19** and continued within the **gostin** project at **t20** can finally become reality.

We basically want to repeat last years hackathon but now using the new subsurface package by implementing it into as many packages as possible. **Let's glue the subsurface-stack together!**

## Goals
* Add `to_subsurface` or `from_subsurface` functionalities to as many packages as possible. (We might brainstorm about the actual name.)
 
## Resources
* Summary of last year's gostin project: https://hackmd.io/@prisae/summary)
* Playlist of recorded meetings: https://youtube.com/playlist?list=PLd9tNwsUm9jN0na4vLXiuP_AJNuLRSsYl

## Skills needed (if any)
At least initially we need folks who are familiar with a package and can implement the required routines to them. But this will also require testing, documentation, building case studies etc, so there is a broad variety of useful skills in this hack.

Even just ideas of what people would like to connect to subsurface is very useful.

* Geoscience
* Programming
* Testing
* Documentation

**Comment by AlexanderJuestel** [comment]
That fits perfectly to #4 and GemGIS ;) 

Looking forward to explore and implement the different capabilities!

**Reply by prisae** [reply]
Indeed, looking forward to it!

**Comment by filippo82** [comment]
[T21 Hackathon Idea Mashup](https://softwareunderground.org/events/2021/4/7/t21-hackathon-idea-mashup) today (or tomorrow depending on your time zone) Wednesday April 14th at 16:00 UTC. Come to Slack to find the link for the meeting!
